projects/plugins/jetpack/modules/masterbar/admin-menu/class-jetpack-admin-menu.php

```diff
@@ -214,7 +214,24 @@ public function add_options_menu() {
 		add_submenu_page( $slug, esc_attr__( 'Writing', 'jetpack' ), __( 'Writing', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/writing/' . $this->domain );
 		add_submenu_page( $slug, esc_attr__( 'Discussion', 'jetpack' ), __( 'Discussion', 'jetpack' ), 'manage_options', 'https://wordpress.com/settings/discussion/' . $this->domain );
 
-		$has_scan     = \Jetpack_Plan::supports( 'scan' );
+		$plan_supports_scan = \Jetpack_Plan::supports( 'scan' );
+
+		if ( ! $plan_supports_scan ) {
+			$products = \Jetpack_Plan::get_products();
+
+			if ( is_array( $products ) ) {
+				$has_scan_product = false;
+
+				foreach ( $products as $product ) {
```

Comment:
Should we add a `$has_scan_product` fallback when `$products` isn't an array, to avoid notices later?